extensions/iceberg/src/main/java/io/deephaven/iceberg/location/IcebergTableLocationFactory.java

```diff
@@ -0,0 +1,30 @@
+//
+// Copyright (c) 2016-2024 Deephaven Data Labs and Patent Pending
+//
+package io.deephaven.iceberg.location;
+
+import io.deephaven.engine.table.impl.locations.TableKey;
+import io.deephaven.engine.table.impl.locations.TableLocation;
+import io.deephaven.engine.table.impl.locations.impl.TableLocationFactory;
+import io.deephaven.engine.table.impl.locations.util.TableDataRefreshService;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * {@link TableLocationFactory} for {@link IcebergTableLocation}s.
+ */
+public final class IcebergTableLocationFactory implements TableLocationFactory<TableKey, IcebergTableLocationKey> {
+    private final Object readInstructions;
+
+    public IcebergTableLocationFactory(@NotNull final Object readInstructions) {
+        this.readInstructions = readInstructions;
```

Comment:
We *can* get these from the `TableLocationKey`. You're following the existing pattern from `PartitionTableLocationFactory`.